packages/material-ui-codemod/README.md

```diff
@@ -44,6 +44,36 @@ This codemod tries to perform a basic expression simplification which can be imp
 // Limitation
 -const spacing = theme.spacing.unit * 5 * 5;
 +const spacing = theme.spacing(5) * 5;
+
+#### `optimal-imports`
+
+Converts all `@material-ui/core` imports more than 1 level deep to the optimal
+form for tree shaking:
+
+```diff
+-import withStyles from '@material-ui/core/styles/withStyles';
+-import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
++import {withStyles, createMuiTheme} from '@material-ui/core/styles';
+```
+
+```sh
+find src -name '*.js' -print | xargs jscodeshift -t node_modules/@material-ui/codemod/lib/v4.0.0/optimal-imports.js
+```
+
+### v3.0.0
```

Comment:
v4.0.0?

Reply:
I don't understand, there are now two codemods in this PR, one under v3, one under v4

Reply:
My concern is with introducing new features to the v3 branch. I believe that the best strategy, for everybody, is to provide all the nice new features in the active development branch. The fewer fragmentation, the better.

Reply:
Oh sorry for the confusion, I switched the target branch back to master.  So even though this would be intended primarily for v3 users it's going into the master branch.

Reply:
It should target the master branch, we all agree.
Now, my point is that we shouldn't provide this new tool for v3 users.

Reply:
Does tree shaking work with top-level imports in v4?  I figure at least some v4 users might want this.  For instance, it's easier to type in additional imports with the top-level form.

Reply:
Yes, it does work for v4 users :)

Reply:
So please, remove the v3 logic. We provide limited support to v3 users. We handle security fixes, live documentation and really import bug fixes.

Reply:
okay, so you mean just move the `top-level-imports` codemod to the v4 folder?

Reply:
Yes, I think that it would be ideal.